src/bin/main.rs

```diff
@@ -61,42 +67,71 @@ fn make_app<'a, 'b>() -> App<'a, 'b> {
     App::new("Howser")
         .about("Document conformity validator for the Rx spec.")
         .version(crate_version!())
-        .arg(
-            Arg::with_name("verbose")
-                .short("v")
-                .long("verbose")
-                .help("Use verbose (multiline) output for errors and warnings."),
-        )
+        .version_message("Prints version information.")
+        .help_message("Prints help information.")
+        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .setting(AppSettings::VersionlessSubcommands)
+        .setting(AppSettings::DisableHelpSubcommand)
         .subcommand(
             SubCommand::with_name("check")
-                .about("Checks that a document intended for use as a prescription conforms to the Rx spec.")
+                .about("Verifies that an .rx file conforms to the Rx spec.")
+                .help_message("Prints help information.")
+                .setting(AppSettings::ArgRequiredElseHelp)
+                .arg(
+                    Arg::with_name("verbose")
+                        .short("v")
+                        .long("verbose")
+                        .help("Use verbose (multiline) output for errors and warnings."),
+                )
                 .arg(
                     Arg::with_name("prescription")
                         .required(true)
-                        .help("Prescription file to check.")
+                        .help("Prescription file to check")
                         .takes_value(true)
                         .value_name("PRESCRIPTION"),
                 ),
         )
         .subcommand(
             SubCommand::with_name("validate")
-                .about("Validate a Markdown document against an Rx Prescription file.")
+                .about("Validates a Markdown document against an .rx Prescription file.")
+                .help_message("Prints help information.")
+                .setting(AppSettings::ArgRequiredElseHelp)
+                .arg(
+                    Arg::with_name("pharmacy")
```

Comment:
I'm not certain whether there are competing design requirements that lead to the current layout, but if it's not too much trouble, I would recommend the use of subcommands rather than an argument to represent the `pharmacy` feature.  

```
howser check README.rx
howser validate README.rx README.md

howser pharmacy check Pharmacy.toml
howser pharmacy validate Pharmacy.toml
```


Adding `--pharmacy` as an argument to `validate` and `check` results in treating one of two other parameters (the target file) to a `validate` call as optional, where previously it was simply required.

In other words, my preference as a downstream user is to have known requirements for each portion of the command tree, rather than having to cross-compare various optional arguments to see which are relevant for each use case 

(see also: fail-early, which could be scoped entirely to the pharmacy subcommands, but currently pollutes the top-level documentation)

Reply:
I think those are good ideas. I was following the existing design requirements, but I think they are flexible.